api/agent/agent.go

```diff
@@ -264,12 +264,18 @@ func (a *agent) getSlot(ctx context.Context, call *call) (Slot, error) {
 
 	isHot := protocol.IsStreamable(protocol.Protocol(call.Format))
 	if isHot {
+
 		// For hot requests, we use a long lived slot queue, which we use to manage hot containers
-		call.slots = a.slotMgr.getHotSlotQueue(call)
+		var isNew bool
+		call.slots, isNew = a.slotMgr.getHotSlotQueue(call)
+		if isNew {
+			a.wg.Add(1)
```

Comment:
this seems like it will be brittle to maintain the corresponding `wg.Done()` with this from here. we're already in Submit, which will handle holding up shutdown until after the job is done, which is kind of all we care about except for removing all containers on shutdown. it seems like it would be easier to just add one of these in `runHot`, which as long as we keep things simple should always be called from within the span of a `Submit` so we don't have to worry about races all that much.

Reply:
ok, I'll remove this one, since runHot() really needs to have wg.Done().